### PR TITLE
BFW-752 loading ini from start of wui

### PR DESCRIPTION
**Pull request by michalrudolf**

### CODE REVIEW

File: src/common/eeprom.h [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #include "variant8.h"
 
 #define EEPROM_ADDRESS  0x0500
-#define EEPROM_VERSION  4
+#define EEPROM_VERSION  6
```
Comment (joshymjose) [inline src/common/eeprom.h]: why you change from 4 to 6? what does this version number do?

File: src/common/eeprom.h [hunk 2]
```diff
@@ -52,11 +52,12 @@
     #define EEVAR_CONNECT_IP4   0x18 // X.X.X.X address encoded in uint32
     #define EEVAR_CONNECT_TOKEN 0x19 // 20char string
     #define EEVAR_LAN_HOSTNAME  0x1A // 20char string
+    #define EEVAR_CONNECT_PORT  0x1B // 0 - 65535 number of connect's port
 #endif                               // (EEPROM_FEATURES & EEPROM_FEATURE_LAN)
 
-#define EEVAR__PADDING 0x19 // 1..4 chars, to ensure (DATASIZE % 4 == 0)
+#define EEVAR__PADDING 0x1C // 1..4 chars, to ensure (DATASIZE % 4 == 0)
```
Comment (joshymjose) [inline src/common/eeprom.h:58]: how did you calculate padding

File: src/common/eeprom.h [hunk 3]
```diff
@@ -52,11 +52,12 @@
...
 
-#define EEVAR_CRC32 0x1a // uint32_t crc32 for
+#define EEVAR_CRC32 0x1D // uint32_t crc32 for
```
Comment (joshymjose) [inline src/common/eeprom.h:60]: how did you calculate crc? or how does it work?

File: src/common/ini_handler.c [hunk 4]
```diff
@@ -2,9 +2,17 @@
 
 #include "ini_handler.h"
 #include "ff.h"
+#include "eeprom.h"
+#include "lwip/dhcp.h"
+#include "lwip/netifapi.h"
+#include "dns.h"
```
Comment (joshymjose) [inline src/common/ini_handler.c:8]: not needed now

File: src/common/ini_handler.c [hunk 5]
```diff
@@ -2,9 +2,17 @@
 
 #include "ini_handler.h"
 #include "ff.h"
+#include "eeprom.h"
+#include "lwip/dhcp.h"
+#include "lwip/netifapi.h"
```
Comment (joshymjose) [inline src/common/ini_handler.c:7]: we need to see if this is needed here. ini handler does only loading ini file!

File: src/common/ini_handler.c [hunk 6]
```diff
@@ -40,3 +48,165 @@ uint8_t ini_load_file(ini_handler handler, void *user_struct) {
     }
     return 1;
 }
+
+static void _change_any_to_static(_networkconfig_t * config) {
+    if (netif_is_up(&eth0)) {
+        netifapi_netif_set_down(&eth0);
+    }
+    config->lan_flag |= LAN_EEFLG_TYPE;
+    eeprom_set_var(EEVAR_LAN_FLAG, variant8_ui8(config->lan_flag));
```
Comment (joshymjose) [inline src/common/ini_handler.c:57]: we shall not use eeprom function directly here, check eeprom_loadsave.c

File: src/common/ini_handler.c [hunk 7]
```diff
@@ -40,3 +48,165 @@ uint8_t ini_load_file(ini_handler handler, void *user_struct) {
...
+
+    netifapi_netif_set_addr(&eth0,
+        (const ip4_addr_t *)&(config->lan.addr_ip4),
+        (const ip4_addr_t *)&(config->lan.msk_ip4),
+        (const ip4_addr_t *)&(config->lan.gw_ip4)
+        );
+#ifdef ENABLE_DNS
+    dns_setserver(0, (const ip4_addr_t *)&(config->dns1_ip4));
```
Comment (joshymjose) [inline src/common/ini_handler.c]: everything realed to dns remove from here even if it is commented out

File: src/common/ini_handler.c [hunk 8]
```diff
@@ -40,3 +48,165 @@ uint8_t ini_load_file(ini_handler handler, void *user_struct) {
...
+    dns_setserver(1, (const ip4_addr_t *)&(config->dns2_ip4));
+#endif
+    if (netif_is_link_up(&eth0) && !(config->lan_flag & LAN_EEFLG_ONOFF)) {
+        netifapi_netif_set_up(&eth0);
+    }
+}
+
+static void _change_static_to_dhcp(uint8_t lan_flg) {
+    if (netif_is_up(&eth0)) {
```
Comment (joshymjose) [inline src/common/ini_handler.c:70]: this should not be here

File: src/common/ini_handler.c [hunk 9]
```diff
@@ -2,9 +2,17 @@
...
+#include "lwip/netifapi.h"
+#include "dns.h"
+#include "ini.h"
 
+#define MAX_UINT16  65535
 static const char network_ini_file_name[] = "/lan_settings.ini"; //change -> change msgboxes in screen_lan_settings
 
+static int ini_config_handler(void *user, const char *section, const char *name, const char *value);
+
 uint8_t ini_save_file(const char *ini_save_str) {
```
Comment (joshymjose) [inline src/common/ini_handler.c:16]: save ini is not needed at the moment, should be avoided to reduce the complexity in work

File: src/common/ini_handler.c [hunk 10]
```diff
@@ -40,3 +48,165 @@ uint8_t ini_load_file(ini_handler handler, void *user_struct) {
...
+    lan_flg &= ~LAN_EEFLG_TYPE;
+    eeprom_set_var(EEVAR_LAN_FLAG, variant8_ui8(lan_flg));
+    if (netif_is_link_up(&eth0) && !(lan_flg & LAN_EEFLG_ONOFF)) {
+        netifapi_netif_set_up(&eth0);
+    }
+}
+
+uint8_t load_config_from_ini(void){
+
+    _networkconfig_t tmp_config;
```
Comment (joshymjose) [inline src/common/ini_handler.c:82]: remove everything realed to dns

**Review by joshymjose — CHANGES_REQUESTED** [review]
we want new clean implementation!

File: src/common/eeprom.h (hunk 1 above)
Comment (michalrudolf) [reply src/common/eeprom.h]: This is a mistake. This number should be incremented whenever we add a variable to eeprom. It causes a reset to default values (including the new variable)

File: src/common/eeprom.h (hunk 2 above)
Comment (michalrudolf) [reply src/common/eeprom.h:58]: This is not calculating of a padding, but only padding's ID in eeprom structure. Look at the IDs above it, there was a collision of ID values before I changed it.

File: src/common/eeprom.h (hunk 3 above)
Comment (michalrudolf) [reply src/common/eeprom.h:60]: Same answer as the conversation above

File: src/common/ini_handler.c (hunk 9 above)
Comment (michalrudolf) [reply src/common/ini_handler.c:16]: this function and ini_load_file() is used by screen_lan_settings, it cannot be avoided.

File: src/common/ini_handler.c (hunk 8 above)
Comment (michalrudolf) [reply src/common/ini_handler.c:70]: I agree, there should be a separate file handling all the netif stuff. Until now, the whole solution was in screen_lan_settings, so I had to move* it out of GUI.. 

*copy and paste as you said no touching of screen_lan_settings

File: src/common/ini_handler.c (hunk 5 above)
Comment (michalrudolf) [reply src/common/ini_handler.c:7]: So where should I put eeprom setting and netif setting? when we load ini file, we have to set those ini file's values...
Should I create separate file for this?

File: src/common/ini_handler.c (hunk 6 above)
Comment (michalrudolf) [reply src/common/ini_handler.c:57]: you mean I should use st25dv64k_user_write_bytes(addr, buff, bs) to write in eeprom?

**Pull request closed without merging**